File: app/src/main/java/com/hippo/ehviewer/client/data/TorrentDownloadMessage.java
package com.hippo.ehviewer.client.data;

import android.os.Parcel;
import android.os.Parcelable;

public class TorrentDownloadMessage implements Parcelable {

    public String path;
    public String name;
    public int progress;
    public boolean failed;

    public TorrentDownloadMessage() {
    }

    protected TorrentDownloadMessage(Parcel in) {
        this.failed = in.readByte() != 0;
        this.name = in.readString();
        this.path = in.readString();
        this.progress = in.readInt();
    }

    public static final Creator<TorrentDownloadMessage> CREATOR = new Creator<TorrentDownloadMessage>() {
        @Override
        public TorrentDownloadMessage createFromParcel(Parcel in) {
            return new TorrentDownloadMessage(in);
        }

        @Override
        public TorrentDownloadMessage[] newArray(int size) {
            return new TorrentDownloadMessage[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
        dest.writeString(this.path);
        dest.writeByte((byte)(this.failed ? 1:0));
        dest.writeInt(this.progress);
    }
}

File: app/src/main/java/com/hippo/ehviewer/client/data/TorrentDownloadMessage.kt
package com.hippo.ehviewer.client.data

import android.os.Parcel
import android.os.Parcelable

class TorrentDownloadMessage : Parcelable {
    @JvmField
    var path: String? = null
    @JvmField
    var dir: String? = null
    @JvmField
    var name: String? = null
    @JvmField
    var progress: Int = 0
    @JvmField
    var failed: Boolean = false

    constructor()

    protected constructor(`in`: Parcel) {
        this.failed = `in`.readByte().toInt() != 0
        this.name = `in`.readString()
        this.path = `in`.readString()
        this.dir = `in`.readString()
        this.progress = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.name)
        dest.writeString(this.path)
        dest.writeString(this.dir)
        dest.writeByte((if (this.failed) 1 else 0).toByte())
        dest.writeInt(this.progress)
    }

    companion object CREATOR : Parcelable.Creator<TorrentDownloadMessage> {
        override fun createFromParcel(parcel: Parcel): TorrentDownloadMessage {
            return TorrentDownloadMessage(parcel)
        }

        override fun newArray(size: Int): Array<TorrentDownloadMessage?> {
            return arrayOfNulls(size)
        }
    }
}
